transformation/src/main/scala/org/broadinstitute/monster/encode/transformation/StepRunTransformations.scala

```diff
@@ -0,0 +1,54 @@
+package org.broadinstitute.monster.encode.transformation
+
+import org.broadinstitute.monster.encode.jadeschema.table.StepRun
+import upack.Msg
+
+/** Transformation logic for ENCODE analysis step run objects. */
+object StepRunTransformations {
+  import org.broadinstitute.monster.common.msg.MsgOps
+
+  /** Transform a raw ENCODE analysis step run into our preferred schema. */
+  def transformStepRun(
+    rawStepRun: Msg,
+    rawStepVersion: Msg,
+    rawStep: Msg,
+    rawGeneratedFiles: Iterable[Msg],
+    fileIdToTypeMap: Map[String, FileType]
+  ): StepRun = {
+
+    // get pipeline run id
+    val stepRunId = CommonTransformations.readId(rawStepRun)
+    val pipelineRunId = for {
+      idPair <- PipelineRunTransformations.getPipelineExperimentIdPair(
+        rawStep,
+        rawGeneratedFiles,
+        stepRunId
+      )
+    } yield {
+      PipelineRunTransformations.getPipelineRunId(idPair._1, idPair._2)
+    }
+
+    // branch files
+    val generatedFileArray = rawGeneratedFiles.toArray
+    val generatedFileBranches = FileTransformations.splitFileReferences(
+      generatedFileArray.map(_.read[String]("@id")),
+      fileIdToTypeMap
+    )
+    val usedFileIds = generatedFileArray
+      .flatMap(_.read[Array[String]]("derived_from"))
+      .distinct
```

Comment:
I would:
1. Create a top-level variable for the `generatedFileIds` instead of inlining the ID extraction in the call to `splitFileReferences`
2. Add a `.diff(generatedFileIds)` to the end of this chain to remove intermediate outputs